integration/ops/suite_test.go

```diff
@@ -0,0 +1,54 @@
+package ops_test
+
+import (
+	"testing"
+
+	"github.com/concourse/concourse/integration/internal/flytest"
+	"github.com/stretchr/testify/require"
+)
+
+func setupUpgradeDowngrade(t *testing.T, fly flytest.Cmd) {
+	t.Run("set up test pipeline", func(t *testing.T) {
+		err := fly.Run("set-pipeline", "-p", "test", "-c", "pipelines/smoke-pipeline.yml", "-n")
+		require.NoError(t, err)
+
+		err = fly.Run("unpause-pipeline", "-p", "test")
+		require.NoError(t, err)
+
+		err = fly.Run("trigger-job", "-j", "test/say-hello", "-w")
+		require.NoError(t, err)
+	})
+}
+
+func verifyUpgradeDowngrade(t *testing.T, fly flytest.Cmd) {
+	t.Run("pipeline and build still exists", func(t *testing.T) {
+		err := fly.Run("get-pipeline", "-p", "test")
+		require.NoError(t, err)
+
+		out, err := fly.Output("watch", "-j", "test/say-hello", "--ignore-event-parsing-errors")
+		require.NoError(t, err)
+		require.Contains(t, out, "Hello, world!")
+	})
+
+	t.Run("can still run pipeline builds", func(t *testing.T) {
+		err := fly.Run("trigger-job", "-j", "test/say-hello", "-w")
+		require.NoError(t, err)
+	})
+
+	t.Run("can still run checks", func(t *testing.T) {
+		err := fly.Run("check-resource", "-r", "test/mockery")
+		require.NoError(t, err)
+	})
+
+	t.Run("can still reach the internet", func(t *testing.T) {
```

Comment:
Hm, not sure why but I can't seem to get this test to work. My job...doesn't seem to be able to reach the internet (`unable to resolve host address 'example.com'`). Probably something misconfigured on my end but not sure what

Reply:
Yeah this one's finicky for me too, but in my case it was on my end - WSL2 sets a private IP in `/etc/resolv.conf` which isn't addressable within the container. Putting `8.8.8.8` in there and disabling WSL's auto-configuration fixed it.

Are you on Docker for Mac? You could try upgrading to 3.1.x: https://github.com/concourse/concourse/issues/6038

Reply:
Currently seeing the issue on 3.1.0 - will try to skip the cleanup step so I can prod at the deployment after the test fails